src/PatternkitDrupalTwigLib.php

```diff
@@ -134,53 +146,93 @@ public function getEditor($subtype = NULL,
    *   Array of metadata objects found.
    */
   protected function getRawMetadata() {
-    $id = $this->getId();
-    $it = new RecursiveDirectoryIterator($this->path);
-    $filter = array('json');
-    $metadata = array();
-    /** @var \SplFileInfo $file */
-    foreach (new RecursiveIteratorIterator($it) as $file) {
-      if (!$file->isFile()) {
-        continue;
-      }
-      $file_path = $file->getPath();
-      $dirs = explode(DIRECTORY_SEPARATOR, $file_path);
-      // All JSON schema must be in an 'api' folder at this time.
-      // @todo Add support for custom setups.
-      // @todo Look at a standard for JSON Schema + JSON Sample data.
-      $num_dirs = count($dirs);
-      if ($num_dirs < 2
-        || array_pop($dirs) !== 'api') {
-        continue;
-      }
-      $file_ext = $file->getExtension();
-      if (!in_array(strtolower($file_ext), $filter, TRUE)) {
-        continue;
+    static $metadata;
+
+    // Use static pattern to avoid rebuilding multiple times per request.
+    if (is_null($metadata)) {
```

Comment:
Might be cleaner to replace this with the check-and-return anti-pattern, but may just be my personal preference.
```
if (!is_null($metadata)) {
  return $metadata;
}
$it = new Recurs...
```

Reply:
/shrug